Core v2 docs update (#819)
spatialos/gdk-for-unity: docs/content/gameobject/world-commands.md

## How to send and receive world commands
We provide the following type for sending and receiving world commands:

  * `WorldCommandSender`

> > We do not provide a `WorldCommandReceiver` as the command will be directly handled by the SpatialOS Runtime.

Reviewer: ```suggestion
> We do not provide a `WorldCommandReceiver` as the SpatialOS Runtime directly handles the command.
```

:D

> The `WorldCommandSender` can be injected without any condition. A MonoBehaviour that requires only these will be enabled as soon as the associated GameObject is created.

Reviewer: ```suggestion
The `WorldCommandSender` can be injected without any condition. A MonoBehaviour that requires only this will be enabled as soon as the associated GameObject is created.
```

```csharp
void SendCreateEntityCommand(WorldCommands.CreateEntity.Request request, Action<WorldCommands.CreateEntity.ReceivedResponse> callback = null);
```

Parameters:

  * `WorldCommands.CreateEntity.Request request`: The command request payload.
  * `Action<WorldCommands.CreateEntity.ReceivedResponse> callback`: Optional. A callback that will be called when the command response is received.

[//]: # (TODO - explain what “handling the response based on the information contained in this object” means)

### ReserveEntityIds

Like other commands, a `CreateEntity` command response can time out, but the entity might still be created. As we do not know which ID was used by the SpatialOS Runtime to create this entity, the worker needs to retry the `CreateEntity` command request. This might lead to creating multiple entities as a new entity ID will be used by the SpatialOS Runtime, if the previous `CreateEntity` command did succeed.
To avoid this, the worker can reserve an entity ID before sending the `CreateEntity` command request. Depending on the command response, you should do the following:

>   * The command succeeded: The entity got successfully created

Reviewer 2: ```suggestion
  * The command succeeded: The entity got successfully created.
```

>   * The command failed with the status code `ApplicationError` and the following message: “'Entity reservation failed. The entity with Id <{id}> could not be found. The reservation might have expired or never existed”. If you have used the entity ID that you received from the `ReserveEntityIds` command response for sending the `CreateEntity` commands, then your entity has already been created and you don’t need to retry it anymore.

Reviewer 2: ```suggestion
  * The command failed with the status code `ApplicationError` and the following message: `Entity reservation failed. The entity with Id <{id}> could not be found. The reservation might have expired or never existed`. 
  <br/>If you have used the entity ID that you received from the `ReserveEntityIds` command response for sending the `CreateEntity` commands, then your entity has already been created and you don’t need to retry it anymore.
```
I know you hate BRs but it helps break up the message and the action.

Reviewer: entity Id instead of entity ID?

  * The command timed out or another error appeared: retry the command.
[//]: # (TODO - link to status codes for error messages.)

You can use the `WorldCommandSender.SendReserveEntityIdsCommand` method to send a command request to reserve entity ids. It has the following signature:

```csharp
void SendReserveEntityIdsCommand(WorldCommands.ReserveEntityIds.Request request, Action<WorldCommands.ReserveEntityIds.ReceivedResponse> callback = null);
```

Parameters:

  * `WorldCommands.ReserveEntityIds.Request request`: The command request payload.
  * `Action<WorldCommands.ReserveEntityIds.ReceivedResponse> callback`: Optional. A callback that will be called when the command response is received.

### DeleteEntity

> You can use the `SendDeleteEntityCommand` method to request the deletion of a SpatialOS entity given its SpatialOS entity ID. It has the following signature:

Reviewer: not completely sure, but I think we use ID and Id interchangeably. probably should do a pass to use one or the other

Reviewer 3: I always use ID. Id seems wrong unless referring to the Freudian concept of the part of you that contains instinctive drives.

You can use the `WorldCommandSender.SendReserveEntityIdsCommand` method to send a command request to reserve entity ids. It has the following signature:

```csharp
void SendReserveEntityIdsCommand(WorldCommands.ReserveEntityIds.Request request, Action<WorldCommands.ReserveEntityIds.ReceivedResponse> callback = null);
```

Parameters:

  * `WorldCommands.ReserveEntityIds.Request request`: The command request payload.
  * `Action<WorldCommands.ReserveEntityIds.ReceivedResponse> callback`: Optional. A callback that will be called when the command response is received.

### DeleteEntity

You can use the `SendDeleteEntityCommand` method to request the deletion of a SpatialOS entity given its SpatialOS entity ID. It has the following signature:

```csharp
void SendDeleteEntityCommand(WorldCommands.DeleteEntity.Request request, Action<WorldCommands.DeleteEntity.ReceivedResponse> callback = null)
```

Parameters:

  * `WorldCommands.DeleteEntity.Request request`: The command request payload.
  * `Action<WorldCommands.DeleteEntity.ReceivedResponse> callback`: Optional. A callback that will be called when the command response is received.

> >  Do not manually delete GameObjects representing entities after sending a `DeleteEntity` command. [You should wait until you receive a callback on ]({{urlRoot}}/content/gameobject/linking-spatialos-entities#the-creation-feature-module).

Reviewer 3: Actually you can do this if you want to.
